core/src/subgraph/validation.rs

```diff
@@ -57,8 +57,14 @@ pub fn validate_manifest(
         errors.push(SubgraphManifestValidationError::DataSourceBlockHandlerLimitExceeded)
     }
 
+    let mut network_name = String::from("none");
+    match manifest.network_name() {
+        Ok(n) => network_name = n,
+        Err(e) => errors.push(e),
+    };
+
     if errors.is_empty() {
-        return Ok(manifest);
+        return Ok((network_name, manifest));
```

Comment:
Can we keep this return value/type as it is? We're going to eventually add non-Ethereum subgraphs and is like to avoid too much Ethereum specific information exposed through generic APIs like this one.